pkg/workloads/manager.go

```diff
@@ -189,8 +189,37 @@ func (d *defaultManager) RunWorkload(ctx context.Context, runConfig *runner.RunC
 	return err
 }
 
+func validateWorkload(ctx context.Context, runConfig *runner.RunConfig) error {
```

Comment:
Nitpick: Call this something like `validateSecretParameters` to make it clear. I also want to avoid it being overloaded to do any future validation as well.